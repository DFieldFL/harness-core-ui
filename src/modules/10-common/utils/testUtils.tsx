import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uikit'
import qs from 'qs'

import strings from 'strings/strings.en.yaml'
import { AppStoreProvider, useAppStoreWriter } from 'framework/hooks/useAppStore'
import type { AppStore } from 'framework/types/AppStore'
import { AUTH_ROUTE_PATH_PREFIX, StringsContext } from 'framework/exports'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): Element | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): Element | null => document.querySelector('.bp3-popover-content')

interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, any>
  defaultAppStoreValues?: Partial<AppStore>
}

export const prependAccountPath = (path: string): string => AUTH_ROUTE_PATH_PREFIX + path

const AppStoreInitializer: React.FC<{ defaultAppStoreValues: TestWrapperProps['defaultAppStoreValues'] }> = ({
  children,
  defaultAppStoreValues
}) => {
  const updateAppStore = useAppStoreWriter()
  updateAppStore(defaultAppStoreValues || {})
  return <> {children} </>
}

export const NotFound = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  const { path = '/', pathParams = {}, defaultAppStoreValues, queryParams = {} } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  const history = createMemoryHistory({ initialEntries: [routePath] })

  return (
    <AppStoreProvider>
      <StringsContext.Provider value={strings}>
        <AppStoreInitializer defaultAppStoreValues={defaultAppStoreValues}>
          <Router history={history}>
            <ModalProvider>
              <RestfulProvider base="/">
                <Switch>
                  <Route exact path={path}>
                    {props.children}
                  </Route>
                  <Route>
                    <NotFound />
                  </Route>
                </Switch>
              </RestfulProvider>
            </ModalProvider>
          </Router>
        </AppStoreInitializer>
      </StringsContext.Provider>
    </AppStoreProvider>
  )
}
