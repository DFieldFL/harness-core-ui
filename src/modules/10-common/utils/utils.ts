import type { IconName } from '@wings-software/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}
