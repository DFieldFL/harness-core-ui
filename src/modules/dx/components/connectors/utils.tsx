import React from 'react'
import { Connectors } from 'modules/dx/constants'
import CreateGITConnector from 'modules/dx/components/connectors/CreateConnector/GITConnector/CreateGITConnector'
import CreateK8sConnector from 'modules/dx/components/connectors/CreateConnector/K8sConnector/CreateK8sConnector'
export const getConnectorWizardByType = (type: string, accountId: string, hideLightModal: () => void) => {
  switch (type) {
    case Connectors.KUBERNETES_CLUSTER:
      return <CreateK8sConnector accountId={accountId} hideLightModal={hideLightModal} />
    case Connectors.GIT:
      return <CreateGITConnector />
    default:
      return null
  }
}
