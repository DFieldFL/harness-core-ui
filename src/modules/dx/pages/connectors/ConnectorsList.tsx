import React, { useState } from 'react'
import { Layout, Container, Button } from '@wings-software/uikit'
import { useParams } from 'react-router-dom'
import { useGetConnectorList, ResponseDTOPageConnectorSummaryDTO } from 'services/cd-ng'
import { PageSpinner } from 'modules/common/components/Page/PageSpinner'
import type { UseGetMockData } from 'modules/common/utils/testUtils'
import { ConnectorSetupModal } from '../../modals/ConnectorModal/ConnectorSetupModal'
import ConnectorsListView from './views/ConnectorsListView'
import css from './ConnectorsList.module.scss'

interface ConnectorsListProps {
  mockData?: UseGetMockData<ResponseDTOPageConnectorSummaryDTO>
}

const enum View {
  GRID,
  LIST
}
const ConnectorsList: React.FC<ConnectorsListProps> = ({ mockData }) => {
  const { accountId } = useParams()
  const [view, setView] = useState(View.LIST)
  const [page, setPage] = useState(0)

  const { loading, data, refetch: reloadConnectorList } = useGetConnectorList({
    accountIdentifier: accountId,
    queryParams: { page: page, size: 10 },
    mock: mockData
  })

  return (
    <Layout.Vertical height={'calc(100vh - 64px'}>
      <Container>
        <Layout.Horizontal className={css.header}>
          <Layout.Horizontal inline width="55%">
            <ConnectorSetupModal />
          </Layout.Horizontal>
          <Layout.Horizontal width="45%" className={css.view}>
            {/* <Button
              minimal
              icon="grid-view"
              intent={view === View.GRID ? 'primary' : 'none'}
              onClick={() => {
                setView(View.GRID)
              }}
            /> */}
            <Button
              minimal
              icon="list"
              intent={view === View.LIST ? 'primary' : 'none'}
              onClick={() => {
                setView(View.LIST)
              }}
            />
          </Layout.Horizontal>
        </Layout.Horizontal>
      </Container>
      {!loading ? (
        view === View.LIST && data?.data?.content?.length ? (
          <ConnectorsListView
            data={data?.data}
            reload={reloadConnectorList}
            gotoPage={pageNumber => setPage(pageNumber)}
          />
        ) : null
      ) : (
        <PageSpinner />
      )}
    </Layout.Vertical>
  )
}

export default ConnectorsList
