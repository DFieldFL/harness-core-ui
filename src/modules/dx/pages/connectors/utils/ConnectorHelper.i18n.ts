export default {
  k8sClusterDetails: 'Kubernetes Connector Details',
  gitConnectorDetails: 'GIT Connector Details',
  secretManagerDetails: 'Secret Manager Details',
  LABEL: {
    k8sCluster: 'Kubernetes Connector',
    gitConnector: 'GIT Connector',
    secretManager: 'Secret Manager'
  }
}
