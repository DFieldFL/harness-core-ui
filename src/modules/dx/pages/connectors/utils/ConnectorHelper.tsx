import { Connectors } from 'modules/dx/constants'
import { StringUtils } from 'modules/common/exports'
import { getKubValidationSchema, AuthTypeFields } from '../Forms/KubeFormHelper'
import i18n from './ConnectorHelper.i18n'

export const AuthTypes = {
  CLIENT_KEY_CERT: 'ClientKeyCert',
  USER_PASSWORD: 'UsernamePassword',
  SERVICE_ACCOUNT: 'ServiceAccount',
  OIDC: 'OpenIdConnect'
}

export const getValidationSchemaByType = (type: string) => {
  if (!type) return null
  switch (type) {
    case Connectors.KUBERNETES_CLUSTER:
      return getKubValidationSchema()

    default:
      return null
  }
}

export const getKubInitialValues = () => {
  return {
    type: 'KUBERNETES_CLUSTER',
    name: 'NAME',
    description: '',
    identifier: '',
    tags: [],
    delegateMode: '',
    credentialType: '',
    credential: {
      masterUrl: '',
      manualCredentialType: '',
      manualCredentials: {
        userName: '',
        encryptedPassword: ''
      }
    }
  }
}

export const getHeadingByType = (type: string) => {
  switch (type) {
    case Connectors.KUBERNETES_CLUSTER:
      return i18n.k8sClusterDetails
    case Connectors.GIT:
      return i18n.gitConnectorDetails
    case Connectors.SECRET_MANAGER:
      return i18n.secretManagerDetails
    case Connectors.APP_DYNAMICS:
      return i18n.appDynamicsDetails
    case Connectors.SPLUNK:
      return i18n.splunkConnectorDetails
    case Connectors.DOCKER:
      return i18n.dockerConnectorDetails
    default:
      return null
  }
}

export const getConnectorTextByType = (type: string) => {
  switch (type) {
    case Connectors.KUBERNETES_CLUSTER:
      return i18n.LABEL.k8sCluster
    case Connectors.GIT:
      return i18n.LABEL.gitConnector
    case Connectors.SECRET_MANAGER:
      return i18n.LABEL.secretManager
    case Connectors.APP_DYNAMICS:
      return i18n.LABEL.appdynamics
    case Connectors.SPLUNK:
      return i18n.LABEL.splunk
    case Connectors.DOCKER:
      return i18n.LABEL.docker
    default:
      return null
  }
}

export const getLabelForEncryptedSecret = (field: string) => {
  switch (field) {
    case AuthTypeFields.passwordRef:
      return 'Password'
    case AuthTypeFields.serviceAccountTokenRef:
      return 'Service Account Token'
    case AuthTypeFields.oidcPasswordRef:
      return 'Password'
    case AuthTypeFields.oidcClientIdRef:
      return 'Client ID'
    case AuthTypeFields.oidcSecretRef:
      return 'Client Secret'
    case AuthTypeFields.clientKeyRef:
      return 'Client Key'
    case AuthTypeFields.clientKeyPassphraseRef:
      return 'Client Key Passphrase'
    case AuthTypeFields.clientCertRef:
      return 'Client Certificate'
    default:
      return ''
  }
}

export const getSecretFieldValue = (field: string) => {
  switch (field) {
    case AuthTypeFields.passwordRef:
      return 'passwordRefSecret'
    case AuthTypeFields.serviceAccountTokenRef:
      return 'serviceAccountTokenRefSecret'
    case AuthTypeFields.oidcClientIdRef:
      return 'oidcClientIdRefSecret'
    case AuthTypeFields.oidcPasswordRef:
      return 'oidcPasswordRefSceret'
    case AuthTypeFields.oidcSecretRef:
      return 'oidcSecretRefSecret'
    case AuthTypeFields.clientKeyRef:
      return 'clientKeyRefSecret'
    case AuthTypeFields.clientKeyPassphraseRef:
      return 'clientKeyPassphraseRefSecret'
    case AuthTypeFields.clientCertRef:
      return 'clientCertRefSecret'
    default:
      return ''
  }
}

export const generateDefaultSecretConfig = (name: string, type: string) => {
  return StringUtils.getIdentifierFromName(name || '').concat(type)
}

export const getLabelForAuthType = (type: string) => {
  switch (type) {
    case AuthTypes.USER_PASSWORD:
      return 'Username and password'
    case AuthTypes.SERVICE_ACCOUNT:
      return 'Service Account Token'
    case AuthTypes.OIDC:
      return 'OIDC Token'
    case AuthTypes.CLIENT_KEY_CERT:
      return 'Client Key Certificate'
    default:
      return ''
  }
}
