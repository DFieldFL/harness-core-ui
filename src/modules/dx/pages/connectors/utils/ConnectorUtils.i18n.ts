export default {
  K8sCluster: 'K8sCluster',
  GitConnector: 'Git'
}
