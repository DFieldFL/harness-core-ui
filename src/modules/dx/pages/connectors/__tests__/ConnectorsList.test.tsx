import React from 'react'
import { render, waitFor, queryByText } from '@testing-library/react'
import { TestWrapper } from 'modules/common/utils/testUtils'
import ConnectorsList from '../ConnectorsList'
import mockData from '../__tests__/mockData'

jest.mock('react-timeago', () => () => 'dummy date')

describe('Connectors List', () => {
  test('render', async () => {
    const { container } = render(
      <TestWrapper path="/account/:accountId/resources/connectors" pathParams={{ accountId: 'dummy' }}>
        <ConnectorsList
          mockData={{
            data: mockData as any,
            loading: false
          }}
        />
      </TestWrapper>
    )
    await waitFor(() => queryByText(container, 'dummyid'))
    expect(container).toMatchSnapshot()
  })
})
