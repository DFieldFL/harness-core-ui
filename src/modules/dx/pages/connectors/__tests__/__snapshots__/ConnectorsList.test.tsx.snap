// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connectors List render 1`] = `
<div>
  <div
    class="Layout---vertical--3Szv StyledProps---main--1Wws listPage"
  >
    <div
      class="StyledProps---font--2ipy StyledProps---main--1Wws"
    >
      <div
        class="Layout---horizontal--3rmQ StyledProps---main--1Wws header"
      >
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws StyledProps---inline--1f52"
          style="width: 55%;"
        >
          <span
            class="bp3-popover-wrapper"
          >
            <span
              class="bp3-popover-target"
            >
              <button
                class="bp3-button StyledProps---font--2ipy Button---button--4ZD5 StyledProps---main--1Wws StyledProps---intent-primary--2Xgi"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-plus StyledProps---main--1Wws StyledProps---padding-right-xsmall--3D04"
                  icon="plus"
                >
                  <svg
                    data-icon="plus"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      plus
                    </desc>
                    <path
                      d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp3-button-text"
                >
                  New Connector
                </span>
                <span
                  class="bp3-icon bp3-icon-chevron-down StyledProps---main--1Wws"
                  icon="chevron-down"
                >
                  <svg
                    data-icon="chevron-down"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      chevron-down
                    </desc>
                    <path
                      d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </span>
          </span>
        </div>
        <div
          class="Layout---horizontal--3rmQ StyledProps---main--1Wws view"
          style="width: 45%;"
        >
          <button
            class="bp3-button bp3-minimal StyledProps---font--2ipy Button---button--4ZD5 StyledProps---main--1Wws StyledProps---intent-primary--2Xgi"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-list StyledProps---main--1Wws"
              icon="list"
            >
              <svg
                data-icon="list"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  list
                </desc>
                <path
                  d="M1 3h14c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm14 10H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h14c.55 0 1-.45 1-1s-.45-1-1-1z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="table table"
    >
      <div
        class="header"
        role="row"
      >
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 25%;"
          title="Toggle SortBy"
        >
          CONNECTOR
          <span
            class="bp3-icon bp3-icon-double-caret-vertical StyledProps---main--1Wws StyledProps---padding-left-small--ODrE"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="15"
              viewBox="0 0 16 16"
              width="15"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 25%;"
          title="Toggle SortBy"
        >
          DETAILS
          <span
            class="bp3-icon bp3-icon-double-caret-vertical StyledProps---main--1Wws StyledProps---padding-left-small--ODrE"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="15"
              viewBox="0 0 16 16"
              width="15"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 20%;"
          title="Toggle SortBy"
        >
          LAST ACTIVITY
          <span
            class="bp3-icon bp3-icon-double-caret-vertical StyledProps---main--1Wws StyledProps---padding-left-small--ODrE"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="15"
              viewBox="0 0 16 16"
              width="15"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 25%;"
          title="Toggle SortBy"
        >
          STATUS
          <span
            class="bp3-icon bp3-icon-double-caret-vertical StyledProps---main--1Wws StyledProps---padding-left-small--ODrE"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="15"
              viewBox="0 0 16 16"
              width="15"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 5%;"
        >
          
        </div>
      </div>
      <div
        class="row card clickable"
        role="row"
      >
        <div
          class="cell"
          role="cell"
          style="width: 25%;"
        >
          <div
            class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
          >
            <span
              class="bp3-icon StyledProps---main--1Wws"
            >
              <svg
                height="30"
                viewBox="0 0 14 14"
                width="30"
              >
                <path
                  d="M7 .4c-.1 0-.2 0-.3.1L2 2.7c-.2.1-.4.3-.5.6l-1.2 5c-.1.2 0 .5.1.7l.1.1 3.2 4c.2.2.4.3.7.3h5.2c.3 0 .5-.1.7-.3l3.2-4c.2-.2.2-.5.2-.7l-1.2-5c-.1-.3-.2-.5-.5-.6L7.3.6c0-.1-.2-.2-.3-.2z"
                  fill="#326ce5"
                />
                <path
                  d="M7 2.1c-.2 0-.3.1-.3.3v.9c0 .1-.1.1-.1.1v.1c-.2 0-.3 0-.5.1-.7.2-1.4.5-1.8 1l-.1-.1H4c-.1-.1-.2-.2-.4-.3L3.4 4h-.1c-.1 0-.1-.1-.2-.1s-.2 0-.2.1c-.1.1-.1.3.1.4h.1c.1.1.2.1.2.1.2.1.3.2.4.3 0 0 .1.1.1.2l.1.1c-.5.7-.7 1.6-.6 2.5h-.1l-.1.1c-.1 0-.3.1-.5.1h-.4c-.2 0-.3.2-.2.3 0 .1.2.2.4.2h.1c.1 0 .2-.1.3-.1.2-.1.3-.1.5-.1.1 0 .1 0 .2.1h.1c.3.9.9 1.6 1.6 2l-.1.1v.2c-.1.1-.1.3-.3.5-.1.1-.1.1-.1.2v.1c-.1.2 0 .3.1.4s.3 0 .4-.2v-.1c0-.1.1-.2.1-.3.1-.2.1-.4.2-.5 0 0 .1 0 .1-.1l.1-.1c.6.2 1.4.3 2.1.2.2 0 .3-.1.5-.1l.1.1s.1 0 .1.1c.1.1.1.3.2.5 0 .1 0 .2.1.3v.1c.1.2.2.2.4.2.1-.1.2-.2.1-.4v-.1c0-.1-.1-.1-.1-.2-.1-.2-.2-.3-.2-.4V10.4c.8-.5 1.3-1.2 1.6-2h.1s.1-.1.2-.1.3.1.5.1c.1 0 .2.1.3.1h.1c.2 0 .3-.1.4-.2 0-.1-.1-.3-.2-.3h-.4c-.2 0-.4 0-.5-.1-.1 0-.1-.1-.1-.1h-.1c.1-.4 0-.9-.1-1.3s-.3-.9-.5-1.2l.1-.1c0-.1 0-.1.1-.2s.2-.2.4-.3c.1 0 .2-.1.2-.1h.1c.1-.1.2-.3.1-.4s-.3-.1-.4 0h-.1l-.2.2c-.1.1-.3.3-.4.3H10l-.1.1c-.6-.6-1.4-1-2.3-1.1v-.1l-.1-.1V3v-.4c-.2-.3-.3-.5-.5-.5zm-.3 2.2l-.1 1.5c0 .1-.1.2-.2.2h-.1l-1.2-.9c.4-.4.8-.6 1.4-.8zm.7 0c.6.1 1.2.4 1.7.8l-1.2.8c-.1.1-.3.1-.3 0l-.1-.1zM4.6 5.7l1.1 1c.1.1.1.2 0 .3l-.1.1-1.4.4c-.1-.7 0-1.3.4-1.8zm4.9 0c.2.3.3.6.4.9s.1.6.1.9l-1.4-.4c-.1 0-.2-.2-.2-.3s0-.1.1-.1zm-2.7 1h.4l.3.3-.1.4-.4.2-.4-.2-.1-.4zm1.4 1.2c.1 0 .1 0 0 0l1.5.2c-.2.6-.6 1.1-1.2 1.5l-.6-1.4c.1-.1.1-.2.3-.3zm-2.4 0c.1 0 .2.1.2.2v.2l-.6 1.4c-.5-.3-.9-.8-1.2-1.5zm1.2.6l.2.1.7 1.3c-.1 0-.2.1-.3.1-.5.1-1.1.1-1.6-.1l.7-1.3c.2 0 .2-.1.3-.1z"
                  fill="#fff"
                  stroke="#fff"
                  stroke-width="0.25"
                />
              </svg>
            </span>
            <div>
              <div
                class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
              >
                <p
                  class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-black--2kJn"
                >
                  dummyName
                </p>
                <span
                  class="bp3-popover-wrapper"
                >
                  <span
                    class="bp3-popover-target"
                  >
                    <div
                      class="Layout---horizontal--3rmQ Layout---layout-spacing-xsmall--C3jb StyledProps---main--1Wws StyledProps---flex--2Jz8 StyledProps---flex-align-center-center--2AmX"
                      tabindex="0"
                    >
                      <span
                        class="bp3-icon StyledProps---main--1Wws"
                      >
                        <svg
                          height="15"
                          viewBox="0 0 14 14"
                          width="15"
                        >
                          <path
                            d="M7 0h-.2c-.2 0-.4.1-.6.3l-6 6c-.3.3-.3.8 0 1.1L2 9.1l4.6 4.6c.3.3.8.3 1.1 0l6-6c.2-.1.3-.3.3-.5v-.7-.3-1.3-2c0-.5-.4-1-1-1h-.9V1c0-.5-.4-1-1-1H7zm0 .6h4.2c.2 0 .4.2.4.4v.9h-1.1c-.3-.3-.5-.4-.8-.4s-.4.1-.6.3h-.5c-.2.1-.4.1-.6.3l-6 6s-.1.1-.1.2L.7 6.9c-.1-.1-.1-.2 0-.3l6-6s.1-.1.2-.1c0 .1.1.1.1.1zm1.9 1.8H13c.2 0 .4.2.4.4v4.3s0 .2-.1.2l-6 6c-.1.1-.2.1-.3 0L2.5 8.8c-.1-.1-.1-.2 0-.3l6-6s.1-.1.2-.1zm2.7.9c-.2 0-.5.1-.6.3-.4.4-.4.9 0 1.3s.9.4 1.3 0 .4-.9 0-1.3c-.3-.2-.5-.3-.7-.3zm0 .7c.1 0 .2 0 .2.1.1.1.1.3 0 .4s-.3.1-.4 0-.1-.3 0-.4c0-.1.1-.1.2-.1z"
                            fill="#4d5969"
                          />
                        </svg>
                      </span>
                      <p
                        class="StyledProps---font--2ipy StyledProps---main--1Wws"
                      >
                        1
                      </p>
                    </div>
                  </span>
                </span>
              </div>
              <p
                class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
              >
                Test Account
              </p>
            </div>
          </div>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 25%;"
        >
          <div>
            <p
              class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-grey400--3csv"
            >
              djcjkfdhk
            </p>
          </div>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 20%;"
        >
          <div
            class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
          >
            <span
              class="bp3-icon StyledProps---main--1Wws"
            >
              <svg
                height="16"
                width="16"
              >
                <path
                  d="M8 0C3.538.036-.035 3.644 0 8.053.036 12.427 3.61 16 8 16h.054c2.133-.018 4.142-.853 5.635-2.382 1.493-1.53 2.329-3.538 2.311-5.671C15.965 3.573 12.391 0 8 0zm5.262 8.836h-1.635a.52.52 0 01-.48-.338l-.836-2.276-1.653 5.245a.49.49 0 01-.462.355H8.16a.51.51 0 01-.462-.302L5.796 7.253l-.907 1.351a.5.5 0 01-.427.232H2.667a.516.516 0 010-1.032h1.511L5.44 5.902a.513.513 0 01.48-.23.488.488 0 01.427.301L8.07 9.867l1.725-5.423a.512.512 0 01.48-.355c.23 0 .426.124.498.338l1.208 3.36h1.28c.285 0 .516.23.516.515a.531.531 0 01-.516.534z"
                  fill="#1D76FF"
                />
              </svg>
            </span>
            dummy date
          </div>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 25%;"
        >
          <div
            class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
          />
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 5%;"
        >
          <div
            class="Layout---horizontal--3rmQ StyledProps---main--1Wws layout"
          >
            <span
              class="bp3-popover-wrapper bp3-dark"
            >
              <span
                class="bp3-popover-target"
              >
                <button
                  class="bp3-button bp3-minimal StyledProps---font--2ipy Button---button--4ZD5 Button---iconOnly--_pvt StyledProps---main--1Wws"
                  type="button"
                >
                  <span
                    class="bp3-icon StyledProps---main--1Wws"
                  >
                    <svg
                      height="16"
                      viewBox="0 0 13 3"
                      width="16"
                    >
                      <g
                        fill="#4d5969"
                      >
                        <path
                          d="M1.5 0C2.3 0 3 .7 3 1.5S2.3 3 1.5 3 0 2.3 0 1.5.7 0 1.5 0M11.5 0c.8 0 1.5.7 1.5 1.5S12.3 3 11.5 3 10 2.3 10 1.5 10.7 0 11.5 0M6.5 0C7.3 0 8 .7 8 1.5S7.3 3 6.5 3C5.9 3 5.4 2.5 5 1.5 5.4.5 5.9 0 6.5 0"
                        />
                      </g>
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
